src/components/views/rooms/RoomList.js

```diff
@@ -260,38 +264,41 @@ module.exports = React.createClass({
     }, 500),
 
     // Update which rooms and users should appear in RoomList as dictated by selected tags
-    updateSelectedTagsEntities: function() {
+    updateSelectedTagsRooms: function(dmRoomMap, updatedTags) {
         if (!this.mounted) return;
```

Comment:
is this guard necessary?

Reply:
Yes, otherwise we'd potentially do a setState without it being mounted

Reply:
yes, but when could `updateSelectedTagsRooms` be called when it's not mounted?

Reply:
When either the FilterStore or GroupStore fires an update.

Reply:
well we remove the FilterStore listener on unmount. As discussed elsewhere, we should also remove the GroupStore listeners, but currently don't. So yes this is correct for now.

Reply:
Fixed here: https://github.com/matrix-org/matrix-react-sdk/pull/1664